### ISSUE TITLE
not all of same reads align 

### ISSUE THREAD

**Issue by jeongmeani** [FelixKrueger/Bismark#347]
Hi, again.

i got question again. Thank you for answer before.

on my fastq, there are same sequence reads.
and i align this sequence to hg19.
but only some of reads align.

for example one read.
`ATTTAGGTTGATTTTATGTTTTTGTTATACGATTTTGGGGAAAGAGAGTTTTGTTGTTTGTTTGTGGGATTTGTTT`
a number of this reads is 1438.
but only 74 reads align to chr8;12206812

why does it happen?

Best Regards

Jeongmin

**Comment by FelixKrueger** [comment]
I don't have an immediate answer to this. Could you extract those 1438 reads into a FastQ file and email it to me?

**Comment by jeongmeani** [comment]
Sorry for late reply.

i send you email which has fastq and bam file related to this issue.

Thank you for your helping :)

Best Regards.

Jeongmin

**Comment by FelixKrueger** [comment]
Hi Jeongmin,

Thanks for sending over the files. Just for the record, Bismark was run with the options `--hisat2 --local --score_min L,0,-0.6 --non_directional` agains the old human genome GRCh37 with a Lambda sequence added to it. I did now take a look and found the following:

- Bismark found 100% ambiguous alignments for this sequence against both the GRCh37 and GRCh38 genomes. I continued with the GRCh38 genome as the other human genome is some 7 years out of date.
- the sequence aligns to at least 3 places in the human genome as perfect matches (CIGAR 76M, alignment score 0). Consequently, the sequence should not align at all, and instead gets printed to the `ambig.bam` file when `--ambig_bam` was specified

So the questions is: why would this sequence map at all (which it shouldn't).

**In Bowtie2 mode,**
- the sequence never produces any alignments, not in default (end-to-end) or local mode, and not against either genome (recent or outdated), and also not with very relaxed alignments. Good.

**In HISAT2 mode,**
- no (unique) alignments in end-to-end mode
- no alignments in `--local` mode
- no alignments against the GRCh37 or GRCh38 genome build
- no alignments with `--score_min L,0,-0.6`

**So I guess the bottom line is:** ****

I have no idea why you are getting any alignments at all, but I am convinced that you shouldn't. Here a few guesses/ comments from my side:
1. It seems clear that the sequence should be classed as ambiguous, and apparently sometimes the `--local` mode does weird things and places the sequences **incorrectly** as unique alignments to chr8.
2. This effect might have to do with the version of HISAT2 you are using, or the fact that you are using an outdated version of the human genome, or both. As far as I am aware some sequence may be aligned whereas others are not, and this may have to do with the read name in the read ID (something to do with heuristics, or random seed generation...). This is really something for the HISAT2 developers and not for me.
3. We have written a Blog post about local alignments erroneously adding perceived unique alignments here: https://sequencing.qcfail.com/articles/soft-clipping-of-reads-may-add-potentially-unwanted-alignments-to-repetitive-regions/
4. Do you really have a library that is **non-directional**? There are only very few kits out there that that generate this type of data. If you library is directional, or a PBAT library, the option `--non_directional` may add incorrect alignments (artefacts)
5. Your data doesn't seem to be sequencing of bisulfite converted RNA-seq. Ts there a specific reason why you are performing splice aware mapping with HISAT2? If the data is not spliced, You should consider the option `--no-spliced-alignment` not to introduce additional errors.
6. Do you have a specific reason of allowing 3 times as may non-bisulfite mismatches and indels than the default mode? For polished genomes this is often not required, and will also potentially additional incorrect alignments compared to a very stringent default mode.

So in summary: I am afraid I cannot reproduce your findings in any way (I am using HISAT2 2.1.0), which is a good thing. I hope you also got the gist that you should not be
- relaxing the mapping parameters
- use spliced mapping
- use local alignments
- use non-directional alignments
- use an outdated genome-build

unless you have **very specific, legitimate reasons** to do so. May a command:

```
bismark --genome GRCh38 file_trimmed.fq.gz
```

Would do the job just fine?

Cheers, Felix
